src/main/java/org/cactoos/collection/CollectionOf.java

```diff
@@ -60,13 +71,131 @@ public CollectionOf(final T... array) {
      * @param src An {@link Iterable}
      */
     public CollectionOf(final Iterable<T> src) {
-        super(() -> {
+        this(() -> {
             final Collection<T> list = new LinkedList<>();
-            for (final T item : src) {
-                list.add(item);
-            }
+            src.forEach(list::add);
             return list;
         });
     }
 
+    /**
+     * Ctor.
+     * @param slr The scalar
+     */
+    public CollectionOf(final Scalar<Collection<T>> slr) {
+        this.col = new Unchecked<>(slr);
+    }
+
+    @Override
+    public int size() {
+        return this.col.value().size();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.col.value().isEmpty();
+    }
+
+    @Override
+    public boolean contains(final Object obj) {
+        return this.col.value().contains(obj);
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return this.col.value().iterator();
+    }
+
+    @Override
+    public Object[] toArray() {
+        return this.col.value().toArray();
+    }
+
+    @Override
+    public <X> X[] toArray(final X[] array) {
+        return this.col.value().toArray(array);
+    }
+
+    @Override
+    public boolean add(final T elem) {
+        return this.col.value().add(elem);
+    }
+
+    @Override
+    public boolean remove(final Object obj) {
+        return this.col.value().remove(obj);
+    }
+
+    @Override
+    public boolean containsAll(final Collection<?> other) {
+        return this.col.value().containsAll(other);
+    }
+
+    @Override
+    public boolean addAll(final Collection<? extends T> other) {
+        return this.col.value().addAll(other);
+    }
+
+    @Override
+    public boolean removeAll(final Collection<?> other) {
+        return this.col.value().removeAll(other);
+    }
+
+    @Override
+    public boolean retainAll(final Collection<?> other) {
+        return this.col.value().retainAll(other);
+    }
+
+    @Override
+    public void clear() {
+        this.col.value().clear();
+    }
+
+    @Override
+    public String toString() {
+        return this.col.value().toString();
+    }
+
+    @Override
+    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings("EQ_UNUSUAL")
```

Comment:
@victornoel is possible change `@edu.umd.cs.findbugs.annotations.SuppressFBWarnings("EQ_UNUSUAL")` for `@SuppressFBWarnings("EQ_UNUSUAL")`?